src/wmtk/operations/tri_mesh/EdgeSwapBase.cpp

```diff
@@ -0,0 +1,100 @@
+#include "EdgeSwapBase.hpp"
+#include <wmtk/SimplicialComplex.hpp>
+#include <wmtk/TriMesh.hpp>
+#include <wmtk/invariants/InteriorEdgeInvariant.hpp>
+
+#include "EdgeCollapse.hpp"
+#include "EdgeSplit.hpp"
+
+namespace wmtk::operations {
+void OperationSettings<tri_mesh::EdgeSwapBase>::initialize_invariants(const TriMesh& m)
+{
+    // outdated + is valid tuple
+    invariants = basic_invariant_collection(m);
+    invariants.add(std::make_shared<InteriorEdgeInvariant>(m));
+}
+
+
+namespace tri_mesh {
+EdgeSwapBase::EdgeSwapBase(Mesh& m, const Tuple& t, const OperationSettings<EdgeSwapBase>& settings)
+    : TriMeshOperation(m)
+    , TupleOperation(settings.invariants, t)
+    , m_settings{settings}
+{}
+
+std::string EdgeSwapBase::name() const
+{
+    return "tri_mesh_edge_swap_base";
+}
+
+Tuple EdgeSwapBase::return_tuple() const
+{
+    return m_output_tuple;
+}
+
+bool EdgeSwapBase::execute()
+{
+    // input
+    //    / \
+    //   /   \
+    //  /  f  \
+    // X--->---
+    //  \     /
+    //   \   /
+    //    \ /
+
+    Tuple split_ret;
+    {
+        OperationSettings<tri_mesh::EdgeSplit> op_settings;
```

Comment:
the opsettings should be be members of the swap op rather than created every time - we don't want to initialize invariants in operations